cmd/integration/integration.go

```diff
@@ -243,8 +248,74 @@ func setIntegrationConfiguration(cmd *cobra.Command, IntegrationName string, sin
 		}
 		sumoLogicSpec := ybmclient.NewSumologicTelemetryProviderSpec(installationToken, accessId, accessKey)
 		IntegrationSpec.SetSumologicSpec(*sumoLogicSpec)
+	case ybmclient.TELEMETRYPROVIDERTYPEENUM_GOOGLECLOUD:
+		if !cmd.Flags().Changed("googlecloud-cred-filepath") {
+			return nil, fmt.Errorf("googlecloud-cred-filepath is required for googlecloud sink")
+		}
+		filepath, _ := cmd.Flags().GetString("googlecloud-cred-filepath")
+		jsonFile, err := os.Open(filepath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open file: %s", err)
+		}
+		defer jsonFile.Close()
+
+		// Read the file into a byte array
+		byteValue, err := io.ReadAll(jsonFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read file: %s", err)
+		}
+
+		// Unmarshal the byte array into a map
+		var googlecloudMap map[string]string
+		if err := json.Unmarshal(byteValue, &googlecloudMap); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal JSON: %s", err)
+		}
+
+		credType, valid := googlecloudMap["type"]
+		if !valid {
```

Comment:
Can you replace the if conditions with a for loop on googlecloudMap?

Reply:
It's a map, do you mean iterate on the key,value pairs? I want them in separate variables to pass on